app/page.tsx

```
@@ -365,34 +398,608 @@ export default function HomePage() {
+      setTtsError("Nothing to speak for this message.");
+      return;
+    }
+    if (isSpeaking) { // Prevent starting new TTS if already speaking
+        console.log("[handleSpeakChatMessage] Already speaking, ignoring request.");
+        return;
+    }
+    setIsSpeaking(true);
+    setTtsLoadingMessageId(messageId); // Set loading for this specific message
+    setTtsError(null);
+
+    if (audioRef.current && audioRef.current.src) {
```

Biome (1.9.4)

[error] Change to an optional chain.

Unsafe fix: Change to an optional chain.


(lint/complexity/useOptionalChain)